nav2_costmap_2d/plugins/costmap_filters/binary_filter.cpp

```diff
@@ -73,6 +73,26 @@ void BinaryFilter::initializeFilter(
   node->get_parameter(name_ + "." + "binary_state_topic", binary_state_topic);
   declareParameter("flip_threshold", rclcpp::ParameterValue(50.0));
   node->get_parameter(name_ + "." + "flip_threshold", flip_threshold_);
+  declareParameter("binary_parameters", rclcpp::ParameterValue(std::vector<std::string>()));
+  node->get_parameter(name_ + "." + "binary_parameters", binary_parameters_);
+  declareParameter("change_parameter_timeout", rclcpp::ParameterValue(10));
+  node->get_parameter(name_ + "." + "change_parameter_timeout", change_parameter_timeout_);
+
+  for (std::string param : binary_parameters_) {
+    BinaryParameter param_struct;
+
+    declareParameter(param + "." + "node_name", rclcpp::ParameterValue(""));
```

Comment:
If there's no default, don't set as empty, set as ParameterType::String so that it throws an exception if its not fully specified. You'd need to wrap parameter getting in try/catch to catch this case and log an error + report it

Reply:
Done! Thank you